evaluate/repository.go

```diff
@@ -77,3 +78,56 @@ func Repository(logger *log.Logger, resultPath string, model evalmodel.Model, la
 
 	return repositoryAssessment, problems, nil
 }
+
+// TemporaryRepository creates a temporary repository and initializes a git repo in it.
+func TemporaryRepository(testDataPath string, repositoryPath string) (temporaryRepositoryPath string, err error) {
+	temporaryPath, err := os.MkdirTemp("", "eval-dev-quality")
```

Comment:
This temporary directory is never ever cleaned up: https://pkg.go.dev/os#MkdirTemp

How about returning a cleanup function